msrestazure/azure_operation.py

```diff
@@ -306,8 +280,23 @@ def get_status_from_location(self, response):
         :param requests.Response response: latest REST call response.
         :raises: BadResponse if response has no body and not status 202.
         """
-        self._check_status(response)
-        self._process_status(response)
+        code = response.status_code
+        if code == 201:
+            self.status = "InProgress"
+        elif code == 200 or \
```

Comment:
Earlier it was acceptable if `code` was 202, but not here. Is that on purpose? And since the difference between this check and the one earlier is just what `code` can be by itself, is it worth it to make the check it's own function?